src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -633,7 +633,14 @@ impl<'a, 'tcx> Visitor<'tcx> for Qualifier<'a, 'tcx, 'tcx> {
 
                 // We might have a candidate for promotion.
                 let candidate = Candidate::Ref(location);
-                if !self.qualif.intersects(Qualif::NEVER_PROMOTE) {
+                let mut unpromotable = self.qualif & Qualif::NEVER_PROMOTE;
+
+                // References to statics are allowed, but only in other statics.
```

Comment:
Why only in other statics? Couldn't references to statics always be promoted

Reply:
It's just what we already allow. Must avoid letting in more until miri is in use.